core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -277,4 +278,14 @@ void decrementAddedCount() {
     Statistics getStatistics() {
         return statistics;
     }
+
+    /********************************************************************************************/
+    abstract class BasicChunkIter {
+        abstract boolean hasNext();
+
+        /** Returns the index of the entry that should be returned next by the iterator.
+         ** NONE_NEXT is returned when iterator came to its end.
+         **/
+        abstract int next(ThreadContext ctx);
+    }
```

Comment:
This should be an interface.

Reply:
in this case the methods become public. Is it OK?